src/main/webapp/script.js

```diff
@@ -0,0 +1,34 @@
+function ratePromoCode(tableRow, rating) {
+    var videoUrl = document.getElementById('promoCodeTable').rows[tableRow].cells[0].firstChild.getAttribute("href");
+    console.log(videoUrl);
+    fetch('/promo-code?rating=' + rating + '&videoUrl=' + videoUrl, {method: 'POST'});
+    getCodes();
+}
+
+function getCodes() {
```

Comment:
Add a comment documenting that this function gets the promocodes and also rewrites the entire table.